docs/pages/config.mdx

```diff
@@ -5,14 +5,292 @@ This file needs to be named `mud.config.ts` and be in the same folder as your `f
+    [function selectors](/world/function-selectors)
+  - `openAccess` (optional, default `true`): a `bool`.
+    If set to `false`, access is limited to:
+    - `System`s in the same namespace
+    - Addresses listed in `accessList`
+    - `System`s listed in `accessList`
+    - [Addresses added to the access list after deployment](/world/namespaces-access-control#modifying-access-control).
+  - `accessList`: an array of `string`. Each address in the array will be granted access to this system, allowing them to call it.
+
+- **`excludeSystems`**: an array of `string`: which systems to not deploy, even if their name ends with “System”.
+
+### Global configuration keys (for multi-namespace config files)
+
+In addition to `namespaces`, these fields are also supported.
+
+- **`enums`**: a record of [enumerations](https://solidity-by-example.org/enum/).
+  The keys are the names of the enumerations.
+  The values are arrays of the strings for the values the enumeration can take.
+
+- **`modules`** an array of module definitions: each module definition has a `name`, `root` (optional), and `args` key.
+
+  - `name`: Name of the module to install. The same module can be installed multiple times. This should be the name of the contract file without `.sol` (eg: if the file is named `DopeModule.sol`, the name of the module is `DopeModule`)
+
+  - `root`: whether to create a `root` module or not. `root` modules have access to all tables and are not bound to namespace restrictions.
+
+  - `args`: a list of arguments to be sent to the `install` function of the module. In this array, you can use the function `resolveTableId`. This function will turn a table name from your config into its low-level ID in the World. It is useful to pass references of a table to a module.
+
+- **`codegen`**: a record of code generation options.
+
+  - **`worldInterfaceName`** a `string`: The name of the interface for the `World`.
+    The default value is `IWorld`.
+  - **`worldgenDirectory`** a `string`: The directory for system and world interfaces.
+    The default value is `world`.
+
+- **`deploy`**: a record of deployment options.
+
+  - **`postDeployScript`** a `string`: Script to execute after the deployment is complete.
+    This script must be placed in the forge scripts directory (see `foundry.toml`) and have a `.s.sol` extension.
+    The default is `PostDeploy`.
+
+  - **`deploysDirectory`** a `string`: Which folder to put the deployment artifacts into after deployment.
+    The default is `./deploys`.
+
+  - **`worldsFile`** a `string`: JSON file for the chain to `World` deploy address mapping.
+    The default is `./worlds.json`.
+
+  - **`upgradeableWorldImplementation`** a `bool`: Whether the `World` is to be deployed behind a proxy to [enable upgrades
+    of the core World implementation](/world/upgrades). The default is `false`.
+
+## Single namespace
+
+This is an older config file format which is still supported.
```

Comment:
We should put this section first. We're still going to continue supporting this single-namespace mode for simple apps that may not make use of multiple namespaces, to reduce both config and directory nesting.